### PR TITLE
chore: migrate stale check to use stale github action

### PR DESCRIPTION
**Pull request by electron0zero**
stale check was added in https://github.com/prometheus/prometheus/pull/6826 using the https://github.com/probot/stale to label PRs with `stale` after 60 days of inactivity.

https://github.com/probot/stale is now deprecated and no longer maintained, and the repository archived since May 20, 2023.

They recommend migrating to [the stale action](https://github.com/actions/stale) instead.

This PR migrates the stale check and keeps the same behaviour as old stale workflow, and adds this file to `scripts/sync_repo_files.sh` so that it's synced to other repos.

also see https://github.com/prometheus/blackbox_exporter/pull/1170 for additional context on what prompted this migration.

### CODE REVIEW

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Stale Check
+on:
+  workflow_dispatch: {}
+  schedule:
+    - cron: '0 0 * * *'
```
Comment (SuperQ) [inline .github/workflows/stale.yml:5]: Let's randomize this so it's not at midnight UTC.

```suggestion
    - cron: '16 22 * * *'
```

**Review by SuperQ — CHANGES_REQUESTED** [review]

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  issues: write
+  pull-requests: write
+jobs:
+  stale:
+    # only run on repos in prometheus org, and skip this workflow in forks.
+    if: github.repository_owner == 'prometheus'
+    runs-on: ubuntu-latest
+    steps:
+      # pin to SHA of v9.0.0: https://github.com/actions/stale/releases/tag/v9.0.0
+      - uses: actions/stale@28ca1036281a5e5922ead5184a1bbf96e5fc984e
```
Comment (SuperQ) [inline .github/workflows/stale.yml:16]:
```suggestion
      - uses: actions/stale@28ca1036281a5e5922ead5184a1bbf96e5fc984e # v9.0.0
```

File: scripts/sync_repo_files.sh [hunk 3]
```diff
@@ -37,7 +37,7 @@ if [ -z "${GITHUB_TOKEN}" ]; then
 fi
 
 # List of files that should be synced.
-SYNC_FILES="CODE_OF_CONDUCT.md LICENSE Makefile.common SECURITY.md .yamllint scripts/golangci-lint.yml .github/workflows/scorecards.yml .github/workflows/container_description.yml"
+SYNC_FILES="CODE_OF_CONDUCT.md LICENSE Makefile.common SECURITY.md .yamllint scripts/golangci-lint.yml .github/workflows/scorecards.yml .github/workflows/container_description.yml  .github/workflows/stale.yml"
```
Comment (SuperQ) [inline scripts/sync_repo_files.sh:40]:
```suggestion
SYNC_FILES="CODE_OF_CONDUCT.md LICENSE Makefile.common SECURITY.md .yamllint scripts/golangci-lint.yml .github/workflows/scorecards.yml .github/workflows/container_description.yml .github/workflows/stale.yml"
```

File: .github/workflows/stale.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  workflow_dispatch: {}
+  schedule:
+    - cron: '0 0 * * *'
+permissions:
+  issues: write
+  pull-requests: write
+jobs:
+  stale:
+    # only run on repos in prometheus org, and skip this workflow in forks.
+    if: github.repository_owner == 'prometheus'
```
Comment (SuperQ) [inline .github/workflows/stale.yml:12]: This won't work with `prometheus-community`.

```suggestion
    if: github.repository_owner == 'prometheus' || github.repository_owner == 'prometheus-community' # Don't run this workflow on forks.
```

**Comment by electron0zero** [comment]
@SuperQ added in the review suggestions. :)

**Review by SuperQ — APPROVED** [review]

**Pull request merged**